Commands sometimes get shell-quoted multiple times

I was trying to install the emacs-w3m recipe and found that it was running `./configure --with-emacs\=/usr/bin/emacs`. With a literal backslash in there. This causes an error, because there is no option called `--with-emacs\`. I suspected that the command arguments were getting passed through `shell-quote-argument` more than once, so I defined some advice around it to echo the input and output each time it is called:

```
(defadvice shell-quote-argument (around echo-call activate)
  (setq before-quoting argument)
  ad-do-it
  (setq after-quoting ad-return-value)
  (message "shell-quote-argument: %S => %S" before-quoting after-quoting))
```

Then I set `el-get-verbose` and `el-get-default-process-sync` to t, and ran `(el-get-post-install "emacs-w3m")` to restart the build process. I got the following in *Messages*:

```
el-get-build emacs-w3m
shell-quote-argument: "./configure" => "./configure"
shell-quote-argument: "--with-emacs=/usr/bin/emacs" => "--with-emacs\\=/usr/bin/emacs"
el-get is waiting for "autoconf" to complete
""
el-get: el-get-build emacs-w3m: autoconf ok.
shell-quote-argument: "--with-emacs\\=/usr/bin/emacs" => "--with-emacs\\\\\\=/usr/bin/emacs"
el-get is waiting for "./configure" to complete
"configure: error: invalid package name: emacs\\
"
Entering debugger...
```

So yes, the configure command is getting double-shell-quoted. Or at least, its argument is. I look in the source and I see that `el-get-build` always calls `shell-quote-argument`, and `el-get-start-process-list` calls it when `:shell` is t. I'm trying to fix it now.

--- a/el-get-build.el
+++ b/el-get-build.el
@@ -81,9 +81,9 @@ recursion.
 	 (default-directory (file-name-as-directory wdir))
 	 (process-list
 	  (mapcar (lambda (c)
-		    (let* ((split    (if (stringp c)
-					 (split-string c)
-				       (mapcar 'shell-quote-argument c)))
+		    (let* ((split    (cond ((stringp c) (split-string c))
+					   ((sequencep c) c)
+					   (t (error "Invalid command: %S" c))))
 			   (c        (mapconcat 'identity split " "))
 			   (name     (car split))
 			   (program  (el-get-build-command-program name))

